src/main/webapp/app/exam/manage/exercise-groups/exercise-groups.component.ts

```diff
@@ -251,6 +256,43 @@ export class ExerciseGroupsComponent implements OnInit {
         }
     }
 
+    getFilePatternOfExercise(exercise: Exercise) {
```

Comment:
Using such methods can be problematic, because they are much slower to check for Angular's automatic change detection than accessing the attributes directly. We should be careful here and avoid them!

Reply:
The problem is that the angular template do not support casting types. We need that because the template accesses properties of the various exercise types. Another way would be to use `$any()` in the template but that breaks the point of type checking